src/components/Dialogs/BulkRenameFilesDialog.vue

```diff
@@ -0,0 +1,317 @@
+
+/** parse TreeNode to plain list */
+const parseNode = (node: TreeNode, parentItem: ItemRow | undefined = undefined, indent: number = 0) => {
+  const item: ItemRow = {
+    indent,
+    name: node.name,
+    fullName: node.fullName,
+    parentItem,
+    show: true,
+    folded: false,
+    selected: false,
+    type: node.type,
+    /** keep the corresponding node, for expand/collapse folder */
+    node
+  }
+  items.push(item)
+  if (node.type === 'folder') {
+    for (const child of node.children) {
+      parseNode(child, item, indent + 1)
+    }
+  }
+}
+
+const toggleFolderFolded = (item: ItemRow, folded: boolean) => {
+  item.folded = folded
+  ;(item as ItemRow<TreeFolder>).node.children.forEach(node => {
+    const correspondence = items.find(item => item.node === node)!
+    correspondence.show = !folded
+    if (correspondence.type === 'folder') {
+      if (folded) {
+        toggleFolderFolded(correspondence, folded)
+      }
+    }
+  })
+}
+
+/**
+ * @return
+ *  * -1: not selected
+ *  * 0: indeterminate(folder only)
+ *  * 1: selected
+ */
+const detectIndeterminate = (node: TreeNode): -1 | 0 | 1 => {
+  const correspondence = items.find(item => item.node === node)!
+  if (node.type === 'folder') {
+    let selectedLength = 0
+    let indeterminateLength = 0
+    node.children.forEach(item => {
+      switch (detectIndeterminate(item)) {
+        case 1:
+          selectedLength++
+          break
+        case 0:
+          indeterminateLength++
+          break
+      }
+    })
+    if (selectedLength === 0 && indeterminateLength === 0) {
+      correspondence.selected = false
+      correspondence.indeterminate = false
+      return -1
+    } else if (selectedLength === node.children.length) {
+      correspondence.selected = true
+      correspondence.indeterminate = false
+      return 1
+    } else {
+      correspondence.indeterminate = true
+      return 0
+    }
+  } else {
+    correspondence.indeterminate = false
+    return correspondence.selected ? 1 : -1
+  }
+}
+
+const folderCheckChange = (item: ItemRow) => {
+  ;(item as ItemRow<TreeFolder>).node.children.forEach(child => {
+    const foundRow = items.find(row => row.node === child)
+    if (foundRow) {
+      foundRow.selected = item.selected
+      if (foundRow.selected) {
+        // unfold all children when selected
+        foundRow.show = true
+        foundRow.folded = false
+      }
+      if (foundRow.type === 'folder') {
+        folderCheckChange(foundRow as ItemRow<TreeFolder>)
+      }
+    }
+  })
+  detectIndeterminate(props.node)
+  // unfold when selected
+  if (item.selected) {
+    item.show = true
+    item.folded = false
+  }
+  dryRunRename()
+}
+
+const fileCheckChange = (item: ItemRow) => {
+  detectIndeterminate(props.node)
+  dryRunRename([item])
+}
+
+const dryRunRename = (partialItems?: ItemRow[]) => {
+  const regexp = new RegExp(regexpInput.value, regexpFlagsInput.value.join(''))
```

Comment:
```suggestion
let regexp: RegExp;
try {
  regexp = new RegExp(regexpInput.value, regexpFlagsInput.value.join(''))
} catch {
  return
}
```

@Larsluph Regarding "regex flag throwing errors," do you mean changing it like this?

Reply:
It would prevent basic error for people that don't know these flags.
This looks like a great quick fix.